src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -40,14 +42,17 @@ public class MeterProviderSdk
             MetricProcessor[] metricProcessors)
         {
             this.Resource = resource;
+            this.metrics = new Metric[MaxMetrics];
 
             // TODO: Replace with single CompositeProcessor.
             this.metricProcessors.AddRange(metricProcessors);
 
+            AggregationTemporality temporality = AggregationTemporality.Cumulative;
             foreach (var processor in this.metricProcessors)
             {
                 processor.SetGetMetricFunction(this.Collect);
                 processor.SetParentProvider(this);
+                temporality = processor.GetAggregationTemporality();
```

Comment:
Kinda outside the scope of this PR, but would it make sense to do away with the list of MetricProcessors here since there can only be one. My first glance of this code made me thing the last processor would dictate the temporality, but then I remembered:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/07114a03676379ab987ae58b911af5aa051ac8e2/src/OpenTelemetry/Metrics/MeterProviderBuilderSdk.cs#L71-L76